starfish/pipeline/features/spots/detector/gaussian.py

```diff
@@ -57,66 +61,34 @@ def __init__(
         self.threshold = threshold
         self.overlap = overlap
         self.is_volume = is_volume
-        if isinstance(blobs_stack, ImageStack):
-            self.blobs_stack = blobs_stack
-        else:
-            self.blobs_stack = ImageStack.from_path_or_url(blobs_stack)
-        self.blobs_image: np.ndarray = self.blobs_stack.max_proj(Indices.ROUND, Indices.CH)
-
-        try:
-            self.measurement_function = getattr(np, measurement_type)
-        except AttributeError:
-            raise ValueError(
-                f'measurement_type must be a numpy reduce function such as "max" or "mean". {measurement_type} '
-                f'not found.')
-
-    @staticmethod
-    def _measure_blob_intensity(image, blobs, measurement_function) -> pd.Series:
-
-        def fn(row: pd.Series) -> Number:
-            row = row.astype(int)
-            result = measurement_function(
-                image[
-                    row['z_min']:row['z_max'],
-                    row['y_min']:row['y_max'],
-                    row['x_min']:row['x_max']
-                ]
-            )
-            return result
-
-        return blobs.apply(
-            fn,
-            axis=1
-        )
+        self.measurement_function = self._get_measurement_function(measurement_type)
 
-    def _measure_spot_intensities(
-            self, stack: ImageStack, spot_attributes: pd.DataFrame
-    ) -> IntensityTable:
+    def image_to_spots(self, data_image: Union[np.ndarray, xr.DataArray]) -> SpotAttributes:
```

Comment:
Rename this method, now required by all implementing algorithms. Calls `detect_spots`